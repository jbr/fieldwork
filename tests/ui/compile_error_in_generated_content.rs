#[derive(fieldwork::Fieldwork)]
struct Something {
    #[fieldwork(get, deref = "()")]
    string: String,
}

#[derive(fieldwork::Fieldwork)]
#[fieldwork(get)]
struct SomethingElse {
    #[fieldwork(deref = "()")]
    string: String,
}

fn main() {}
