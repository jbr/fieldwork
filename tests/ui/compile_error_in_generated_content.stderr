error[E0308]: mismatched types
 --> tests/ui/compile_error_in_generated_content.rs:3:17
  |
3 |       #[fieldwork(get, deref = "()")]
  |                   ^----------------
  |                   |
  |  _________________expected `&()` because of return type
  | |
4 | |     string: String,
  | |__________^ expected `&()`, found `&str`
  |
  = note: expected reference `&()`
             found reference `&str`

error[E0308]: mismatched types
  --> tests/ui/compile_error_in_generated_content.rs:11:5
   |
10 |       #[fieldwork(deref = "()")]
   |  _________________________-
11 | |     string: String,
   | |     ^^^^^^^^^^^^^-
   | |_____|____________|
   |       |            expected `&()` because of return type
   |       expected `&()`, found `&str`
   |
   = note: expected reference `&()`
              found reference `&str`
